docs/architecture/adr-020-post-ownership-change.md

```diff
@@ -68,10 +68,10 @@ message PostOwnerTransferRequest {
 To simplify the management of post owner transfer requests, we will store each request using the following key format:
 
 ```
-IncomingPostOwnerTransferRequestPrefix | SubspaceID | ReceiverAddress | PostID | -> Protobuf(PostOwnerTransferRequest)
+PostOwnerTransferRequestPrefix | SubspaceID | PostID | -> Protobuf(PostOwnerTransferRequest)
 ```
 
-This structure enables the receiver to easily manage incoming requests by iterating over all requests with a given subspace ID and receiver address, which will be the most frequently used query.
+This structure enables Desmos to easily manage requests by iterating over all requests with a given subspace ID and post ID when a post is moved or deleted.
```

Comment:
@RiccardoM 
I have identified an issue with this design: we are unable to determine the post's ownership receiver when it is deleted or moved. This leads to a time complexity of O(n) when deleting a post, as it requires the following steps:
- Iterate over all requests by subspace ID.
- Find the post ID of the request that matches the post to be deleted.
- Delete the corresponding request store key.

To address this issue, I have simplified the store key implementation, resulting in an improved time complexity of O(1) for the post deletion process.